gui/canvaspainter/v7/src/TCanvasPainter.cxx

```diff
@@ -198,254 +195,245 @@ class TCanvasPainter : public THttpWSHandler,
 
-   // open new display for the canvas
    virtual void NewDisplay(const std::string &where) override;
 
-   // void ReactToSocketNews(...) override { SendCanvas(); }
-
    /** \class CanvasPainterGenerator
-       Creates TCanvasPainter objects.
-     */
+          Creates TCanvasPainter objects.
+        */
 
    class GeneratorImpl : public Generator {
-   public:
-      /// Create a new TCanvasPainter to paint the given TCanvas.
-      std::unique_ptr<TVirtualCanvasPainter> Create(const ROOT::Experimental::TCanvas &canv,
-                                                    bool batch_mode) const override
-      {
-         return std::make_unique<TCanvasPainter>("name", canv, batch_mode);
-      }
-      ~GeneratorImpl() = default;
-
-      /// Set TVirtualCanvasPainter::fgGenerator to a new GeneratorImpl object.
-      static void SetGlobalPainter()
-      {
-         if (TVirtualCanvasPainter::fgGenerator) {
-            R__ERROR_HERE("CanvasPainter") << "Generator is already set! Skipping second initialization.";
-            return;
+      public:
+         /// Create a new TCanvasPainter to paint the given TCanvas.
+         std::unique_ptr<TVirtualCanvasPainter> Create(const ROOT::Experimental::TCanvas &canv, bool batch_mode) const override
+         {
+            return std::make_unique<TCanvasPainter>(canv, batch_mode);
+         }
+         ~GeneratorImpl() = default;
+
+         /// Set TVirtualCanvasPainter::fgGenerator to a new GeneratorImpl object.
+         static void SetGlobalPainter()
+         {
+            if (TVirtualCanvasPainter::fgGenerator) {
+               R__ERROR_HERE("NewPainter") << "Generator is already set! Skipping second initialization.";
+               return;
+            }
+            TVirtualCanvasPainter::fgGenerator.reset(new GeneratorImpl());
          }
-         TVirtualCanvasPainter::fgGenerator.reset(new GeneratorImpl());
-      }
 
-      /// Release the GeneratorImpl object.
-      static void ResetGlobalPainter() { TVirtualCanvasPainter::fgGenerator.reset(); }
-   };
-};
+         /// Release the GeneratorImpl object.
+         static void ResetGlobalPainter() { TVirtualCanvasPainter::fgGenerator.reset(); }
+      };
 
-TString TCanvasPainter::fAddr = "";
-THttpServer *TCanvasPainter::gServer = 0;
+};
 
-/** \class TCanvasPainterReg
-  Registers TCanvasPainterGenerator as generator with ROOT::Experimental::Internal::TVirtualCanvasPainter.
-  */
-struct TCanvasPainterReg {
-   TCanvasPainterReg() { TCanvasPainter::GeneratorImpl::SetGlobalPainter(); }
-   ~TCanvasPainterReg() { TCanvasPainter::GeneratorImpl::ResetGlobalPainter(); }
-} canvasPainterReg;
+struct TNewCanvasPainterReg {
+   TNewCanvasPainterReg() { TCanvasPainter::GeneratorImpl::SetGlobalPainter(); }
+   ~TNewCanvasPainterReg() { TCanvasPainter::GeneratorImpl::ResetGlobalPainter(); }
+} newCanvasPainterReg;
 
-/// \}
 
-} // unnamed namespace
+} // namespace Experimental
+} // namespace ROOT
 
```

Comment:
Couldn't you keep the namespaces until the end? That would save you to use long names like: `ROOT::Experimental::TCanvasPainter` every time

Reply:
It is like @Axel-Naumann is doing in all other parts of v7 code.
On the long run `::Experimental` namespace will disappear and name `ROOT::TCanvasPainter` should be ok.

Reply:
OK, fine with me then